Randomize ColumnID test

RichardJCai opened the pull request:
Seeing what breaks when ColumnIDs are randomized.

RichardJCai committed:
Randomize ColumnID test

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46628)
<!-- Reviewable:end -->

RichardJCai committed:
Fix ordinal ppos in information_schema.columns

RichardJCai committed:
Fix seed for randomizing col ids

RichardJCai committed:
change adnum to use col id

RichardJCai committed:
Revert "Fix ordinal ppos in information_schema.columns"

This reverts commit 7a6ce945bbffca950071137f4f87699e2bab8ef0.

Release justification:
Release note (<category, see below>): <what> <show> <why>

RichardJCai committed:
Add Logical Column ID field to ColumnDescriptor

Release note (<category, see below>): <what> <show> <why>

Release justification:

RichardJCai committed:
Correct LogicalColumnID with out of order ColumnID

Release justification:
Release note (<category, see below>): <what> <show> <why>
